lib/facility.rb

```diff
@@ -1,14 +1,42 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :facility, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility)
+    @name = facility[:name]
+    @address = facility[:address]
+    @phone = facility[:phone]
     @services = []
+    @facility = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+  
+  def register_vehicle(vehicle)
+    registered_vehicles << vehicle
+    vehicle.new_registration_date
+    
+    if vehicle.antique?
+      @collected_fees += 25 
+      vehicle.plate_type = :antique
+    elsif vehicle.electric_vehicle?
+      @collected_fees += 200
+      vehicle.plate_type = :ev
+    else 
+      @collected_fees += 100
+      vehicle.plate_type = :regular
+    end
+  end
+
+  # def administer_written_test
+  #   permit? == true
+  #   age >= 16
+  # end
 end
```

Comment:
You're close with this method! I know you mentioned this was a really tough one for you. We only want to register a vehicle IF the list of services contains "Vehicle Registration". So, we'll want to add some type of conditional into this method such that IF "Vehicle Registration" is in the list of services, we do everything between lines 19 and 31, otherwise, we do nothing.